### ISSUE TITLE
Key `api` is no longer available in RowDataUpdatedEvent 

### ISSUE THREAD

**Issue by Miniplop** [ag-grid/ag-grid#5628]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Current behavior**

When using `onRowDataUpdated` the event receive in the callback have only `type` key

We have a `onRowUpdated` callback function that make post edition computation. Then we updated from  aggrid 28.1.0 to aggrid 28.1.1, and we noticed that the `event` given to the callback only contained a `type` key.  The type `RowDataUpdatedEvent` and the [documentation](https://www.ag-grid.com/react-data-grid/grid-events/#reference-clientRowModel-rowDataUpdated) still mention that this event should receive the `gripApi`. 

 An overview of what we have in our code 
```
 const onRowDataUpdated = (event: GridReadyEvent) => {
    event.api.forEachNode((node) => {
      ....
    });
  };

<AgGridReact
          ....
          onRowDataUpdated={onRowDataUpdated}
         ....
/>
```

**Expected behavior**
We expected to found the `api` key in the event. For now we quick fixed the problem by using an gridApi external reference. I think the documentation + types should be updated if it's not a regression. 

Thank you for your help

**Please tell us about your environment:**
node 16.14.2
react : 18.2.0
@ag-grid-community/react:  28.1.1

* **AG Grid version:** X.X.X
@ag-grid-community/react:  28.1.1

* **Browser:**
Chrome

* **Language:** 
TypeScript 4.8.3

**Comment by seanlandsman** [comment]
Hi

Are you talking about the types here (ie what's on `GridReadyEvent`) or what's available at runtime?

If we're talking about runtime I can see the following available when this callback is called:

![image](https://user-images.githubusercontent.com/1329081/193276436-424b450c-bd3f-448d-b549-5f76146774e7.png)

Please see this plunker as an example: https://plnkr.co/edit/A4CmAQEf7Z9mOBN8?open=index.jsx

If you're talking about types you really should be using `RowDataUpdatedEvent`, but in any event both `RowDataUpdatedEvent` and `GridReadyEvent` should have both `api` and `columnApi` available on these types.

**Comment by Miniplop** [comment]
> Hi
> 
> Are you talking about the types here (ie what's on `GridReadyEvent`) or what's available at runtime?
> 

I'm talking about what's available at runtime. It works well with the previous "28.1.0" agGrid event but not with the version "28.1.1"

Don't know why it stops working

**Comment by seanlandsman** [comment]
If you look at the plunker above - which uses 28.1.1 - you should see that the api is available on the parameter passed in.

Are you able to supply a reproducible example of this issue so that we can investigate?

**Comment by Miniplop** [comment]
Hello @seanlandsman, I found a good lead :)

I have checked the project dependency and I noticed that all the aggrid packages are at "28.1.1" version except one "@ag-grid-community/core" (see the screenshot)

<img width="365" alt="image" src="https://user-images.githubusercontent.com/15800180/194033280-7c24d022-1494-465a-8059-770273a7ae20.png">

After investigation I found that the package "@ag-grid-community/core@28.1.0" is coming from the package dependencies of "@ag-grid-enterprise/column-tool-panel@28.1.1" which is a dependency of "@ag-grid-enterprise/menu@28.1.1"

So naturally I try to update all the packages to a commun version "28.2.0" (see screenshot)

<img width="385" alt="image" src="https://user-images.githubusercontent.com/15800180/194034059-7f48a044-c9b8-4e1a-be18-e304fe64690a.png">

And ..... 🥁  The event object is now filled with all expected informations 

![image](https://user-images.githubusercontent.com/15800180/194034498-12f49f10-4b6a-439c-ba4e-0e17b53a922d.png)
 
To sum up, I think when I updated the package `@ag-grid-enterprise/menu` from "28.1.0" to "28.1.1" it didn't updated the core dependency from "28.1.0" to "28.1.1". Don't know why it didn't happen. I tried to install the package '@ag-grid-enterprise/menu@28.1.1' to a fresh new react project and it installed the "28.1.1" version of the core so it doesn't look like an ongoing issue to me.

**Comment by StephenCooper** [comment]
Ok, so ensuring the matching versions were installed resolved the issue. Thanks for providing your resolution.
